### PR TITLE
Search box is not aligned to the index nav width on the release notes page

### LINKED ISSUE #2045: Search overlay not closing after selecting item
If the user is on a certain page and uses the Algolia search functionality to search for any term that links to the same page, when he clicks the search result, the overlay doesn't disappear.

It happens on production using Algolia search. Here's a video: https://commercetools.slack.com/files/U020GRFMX8E/F07EKBKU7A9/docs_search_grey_overlay.gif

Steps to reproduce:
- Navigate to https://docs.commercetools.com/api/projects/inventory
- Click in the search box and write `inventory add` click on the `Update Actions > Add Quantity` item
- Observe

Expected result:
- The page scrolls to the relevant section
- The overlay disappears

Actual result:
- The page scrolls to the relevant section
- The overlay stays

### LINKED ISSUE #2068: Searchbox is not aligned to the index nav width on the release notes page
As you can see, the search box is not aligned with the index nav. We should align the searchbox the same way as in a normal content page. The AI icon should be on the left side of the index nav border.

Screenshot 2024-09-03 at 11 47 59

### PR DESCRIPTION
**Pull request by ColinRosati**
**Description of changes**
- Align layout header and layout page content for release notes
- util for shared layout for header and page content
- refactor grid layouts 
  - Shared const IDs in layout components
  - extracted config and logic

**Link to original ticket**
closes #2068 

**Link to change** (deep-link for reviewing the change)

**Screenshot of changes (if applicable)**

[DoD guidelines](https://commercetools.atlassian.net/wiki/spaces/ET/pages/738820530/Definition+of+Done)

- [ ] user documentation added?
- [ ] stakeholders approve changes? (if needed)

### CODE REVIEW

File: packages/gatsby-theme-docs/src/layouts/internals/layout-design-config.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * |---COLUMN_ONE--------|--COLUMN_TWO--|
+ * |---COLUMN_ONE-----------------------|
+ *
+ * |---COLUMN_ONE-----------------------|
+ * |---COLUMN_ONE-----------------------|
+ *
+ * Note: Release notes page has a different varient on column two's width to account for search filters
+ */
+export const contentLayoutConfig = {
```
Comment (ColinRosati) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-design-config.ts:24]: Extracted a single source of truth for layout config mapping.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-design-config.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      designSystem.dimensions.widths.releaseNoteFilters,
+    column2: `minmax(
+          ${designSystem.dimensions.widths.pageNavigationSmall},
+          ${designSystem.dimensions.widths.releaseNoteFilters}
+        )`,
+  },
+};
+
+// Namespace declarations for all app layout grid ids
+export const GRID_ID_ROW_ONE = {
```
Comment (ColinRosati) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-design-config.ts:62]: Extracting shared vars for global grid layout ids

File: packages/gatsby-theme-docs/src/layouts/internals/layout-design-utils.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+/**
+ * getPageLayoutGridStyles utility is a helper function
+ * to share layout styles across multiple components
+ *
+ * @param gridRows string of locally defined grid rows
+ * @param isReleaseNotesPage boolean to determine if the page is a release notes page
+ * @param allowWideContentLayout boolean to determine if the page should allow wide content layout
+ * @returns
+ */
+export const getPageLayoutGridStyles = ({
```
Comment (ColinRosati) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-design-utils.ts:13]: This is the single entry point for layout components to get their grid styling from. We can avoid layout drift by reducing the duplicated code

File: packages/gatsby-theme-docs/src/layouts/internals/layout-header.js [hunk 4]
```diff
@@ -26,65 +36,17 @@ const Container = styled.header`
...
-        ${designSystem.dimensions.widths.pageContentWideWithMarginsMax}
-      )
-      minmax(${designSystem.dimensions.widths.pageNavigationSmall}, ${designSystem.dimensions.widths.pageNavigation});
-  }`
-      : ''}
+    getPageLayoutGridStyles({
+      gridRows,
+      isReleaseNotesPage: props.isReleaseNotesPage,
+      allowWideContentLayout: props.allowWideContentLayout,
+      isHeader: true,
```
Comment (ColinRosati) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:44]: Header flag configures fixed width or not

File: websites/self-learning-smoke-test/src/layouts/learning-home.js [hunk 5]
```diff
@@ -60,8 +64,8 @@ const HeaderContainer = styled.div`
   /* Override some styles of the original <LayoutHeader> */
   header {
     grid:
-      [row1-start] 'header-top-menu header-searchbox' ${designSystem.dimensions
-        .heights.header} [row1-end]
+      [row1-start] '${GRID_ID_HEADER_TOP_MENU} ${GRID_ID_HEADER_SEARCHBOX}' ${designSystem
```
Comment (gabriele-ct) [inline websites/self-learning-smoke-test/src/layouts/learning-home.js:67]: Is this the actual fix? @ColinRosati

**Review by gabriele-ct — APPROVED** [review]
The refactor seems ok, I'm relying mostly on VRT to ensure possible side effects.
One important thing @ColinRosati make sure to link the correct issue to the PR, as far as I can see the one linked is not related with the task

File: websites/self-learning-smoke-test/src/layouts/learning-home.js (hunk 5 above)
Comment (ColinRosati) [reply websites/self-learning-smoke-test/src/layouts/learning-home.js:67]: Theres few files where I remove the hard coded text for the shared vars. This is just a quality of life refactor. Not the functional fix

**Comment by ColinRosati** [comment]
> The refactor seems ok, I'm relying mostly on VRT to ensure possible side effects. One important thing @ColinRosati make sure to link the correct issue to the PR, as far as I can see the one linked is not related with the task

Thanks updated the link. I was relying on VRT as well. I manually tested few different layouts

File: websites/self-learning-smoke-test/src/layouts/learning-home.js (hunk 5 above)
Comment (gabriele-ct) [reply websites/self-learning-smoke-test/src/layouts/learning-home.js:67]: ah, yes, my bad. Thank you

**Comment by zbalek** [comment]
> Hey @zbalek can you help check that this fixes the layout issue for release notes? I also did some refactor to unify layout code could you click around a few pages to see if you can spot any layout changes?

Hi @ColinRosati can you please share a preview link?

**Comment by ColinRosati** [comment]
@zbalek Theres a few different layouts here:
 https://docskit-cc-searchbox-is-not-aligned-to-the-index-nav-width-o.commercetools.vercel.app/docs-smoke-test/
https://docskit-cc-searchbox-is-not-aligned-to-the-index-nav-width-o.commercetools.vercel.app/api-docs-smoke-test/

For the release notes search box I can only test that locally unfornately and will have to wait till we merge and test the changes in docs release

**Comment by zbalek** [comment]
@ColinRosati Okay, in that case I only took a look at the layouts in the links you've shared. The second one actually misses a search box completely (?🤔)

image

**Comment by ColinRosati** [comment]
@zbalek  from what I see if we compare against main on that one route the search box is not present.

Comparing
https://docs-kit.commercetools.vercel.app/api-docs-smoke-test/layouts/type-regular-desktop
https://docskit-cc-searchbox-is-not-aligned-to-the-index-nav-width-o.commercetools.vercel.app/api-docs-smoke-test/layouts/type-regular-desktop

Maybe Im missing something though?

**Comment by gabriele-ct** [comment]
As fare as I can see, the ticket seems to refer to the release notes page which exists only on the docs repo and uses a custom layout see `websites/docs/src/pages/release-notes.js` in `docs` repo.
I'd start from there.

**Comment by ColinRosati** [comment]
> As fare as I can see, the ticket seems to refer to the release notes page which exists only on the docs repo and uses a custom layout see `websites/docs/src/pages/release-notes.js` in `docs` repo. I'd start from there.

Thats right, the docs release-notes page as well as our other pages use the layout components within docs-kit. Ive tested the changes locally but unfortunately wont be able to get a preview deploy of the docs repo. I would propose we move forward with the PR and do the feature QA in docs release

**Comment by gabriele-ct** [comment]
In this case I would proceed as follow:
- merge this PR
- create a new release that includes these changes
- apply the new release to docs repo on a feature branch
- ensure the issue with the release-notes page is gone, or work on fix for it (in the same branch)
- finally ask stakeholders to verify the issue directly on the docs repo preview.
- merge the docs PR

Ensure that the [issue](https://github.com/commercetools/commercetools-docs-kit/issues/2068) remains open as this PR it's not going to fix it directly.

@FFawzy do you agree?

**Pull request merged**